packages/compass-sidebar/src/components/connections-filter-popover.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useCallback, type PropsWithChildren } from 'react';
+
+import {
+  css,
+  Icon,
+  IconButton,
+  InteractivePopover,
+  Label,
+  Overline,
+  palette,
+  spacing,
+  Toggle,
+  Tooltip,
+  useId,
+} from '@mongodb-js/compass-components';
+import type { ConnectionsFilter } from './use-filtered-connections';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'flex-start',
+  padding: spacing[300],
+  minWidth: 270,
+});
+
+const closeButtonStyles = css({
+  // An alternative to this is to pass hideCloseButton to InteractivePopover,
+  // but that throws an error when the popover is opened
+  display: 'none',
+});
+
+const activatedIndicatorStyles = css({
+  position: 'absolute',
+  top: spacing[50],
+  right: spacing[50],
+});
+
+const groupStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  gap: spacing[200],
+  marginTop: spacing[200],
+});
+
+type ConnectionsFilterPopoverProps = PropsWithChildren<{
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  filter: ConnectionsFilter;
+  onFilterChange(
+    updater: (filter: ConnectionsFilter) => ConnectionsFilter
+  ): void;
+}>;
+
+export default function ConnectionsFilterPopover({
+  open,
+  setOpen,
+  filter,
+  onFilterChange,
+}: ConnectionsFilterPopoverProps) {
+  const onExcludeInactiveChange = useCallback(
+    (excludeInactive: boolean) => {
+      onFilterChange((filter) => ({
+        ...filter,
+        excludeInactive,
+      }));
+    },
+    [onFilterChange]
+  );
+
+  const excludeInactiveToggleId = useId('exclude-inactive-toggle');
+  const excludeInactiveLabelId = useId('exclude-inactive-label');
+
+  // Add future filters to the boolean below
+  const isActivated = filter.excludeInactive;
+
+  return (
+    <InteractivePopover
+      open={open}
+      setOpen={setOpen}
+      blurTriggerOnClose
```

Comment:
It's a common accessible behavior to return focus back to the trigger, especially helpful if you're navigating with a keyboard, all leafygreen Menus are doing that and we are following their behavior here with our InteractivePopover. Why are we breaking the pattern here?

Reply:
I added this because bringing back the focus makes the tooltip show even if it's not hoved 🤔 Perhaps I can return focus to something else?

Reply:
For what it's worth, it seems the "tab position" is remembered. I.e. when tabbing after closing the popover you can continue to the connections tree.

Reply:
Your focus is on body though, you don't get notificed by voice over that you're back on button. If we want to really address this, we probably should make sure that the button has a proper (aria-)label and shows tooltip only on hover where it makes sense to show it, not on keyboard focus, but this probably requires figuring this out with leafygreen team. You can probably circumvent this by using an extra sibling element as trigger and not the IconButton itself, this should also work. Or we can just ignore it, we're compromising on accessibility in a lot of other places too already 🤷

Reply:
> we're compromising on accessibility in a lot of other places too already 🤷

Yeah, but I agree we should avoid regressing further on this 👍

I reverted the change to add a `blurTriggerOnClose` prop and instead used `Tooltip` as a controlled component. Thanks a lot for the explanations and the comment in the first place 👍